go/stellar/stellarsvc/frontend.go

```diff
@@ -1253,6 +1253,102 @@ func (s *Server) CreateWalletAccountLocal(ctx context.Context, arg stellar1.Crea
 	return stellar.CreateNewAccount(s.mctx(ctx), arg.Name)
 }
 
+func (s *Server) BuildRequestLocal(ctx context.Context, arg stellar1.BuildRequestLocalArg) (res stellar1.BuildRequestResLocal, err error) {
+	ctx, err, fin := s.Preamble(ctx, preambleArg{
+		RPCName:       "BuildRequestLocal",
+		Err:           &err,
+		RequireWallet: true,
+	})
+	defer fin()
+	if err != nil {
+		return res, err
+	}
+
+	tracer := s.G().CTimeTracer(ctx, "BuildRequestLocal", true)
+	defer tracer.Finish()
+
+	ctx = s.buildPaymentSlot.Use(ctx, arg.SessionID)
+	if err := ctx.Err(); err != nil {
+		return res, err
+	}
+
+	readyChecklist := struct {
+		to         bool
+		amount     bool
+		secretNote bool
+	}{}
+	log := func(format string, args ...interface{}) {
+		s.G().Log.CDebugf(ctx, "brl: "+format, args...)
+	}
+
+	bpc := stellar.GetBuildPaymentCache(s.mctx(ctx))
+	if bpc == nil {
+		return res, fmt.Errorf("missing build payment cache")
+	}
+
+	// -------------------- to --------------------
+
+	tracer.Stage("to")
+	skipRecipient := len(arg.To) == 0
+	if !skipRecipient {
+		_, err := bpc.LookupRecipient(s.mctx(ctx), stellarcommon.RecipientInput(arg.To))
+		if err != nil {
+			log("error with recipient field %v: %v", arg.To, err)
+			res.ToErrMsg = "recipient not found"
+			skipRecipient = true
+		} else {
+			readyChecklist.to = true
+		}
+	}
+
+	// -------------------- amount + asset --------------------
+
+	tracer.Stage("amount + asset")
+	bpaArg := buildPaymentAmountArg{
+		Amount:   arg.Amount,
+		Currency: arg.Currency,
+		Asset:    arg.Asset,
+	}
+
+	// For requests From is always the primary account.
+	primaryAccountID, err := bpc.PrimaryAccount(s.mctx(ctx))
+	if err != nil {
+		log("PrimaryAccount -> err:%v", err)
+		res.Banners = append(res.Banners, stellar1.SendBannerLocal{
+			Level:   "error",
+			Message: "Could not find primary account.",
+		})
+	} else {
+		bpaArg.From = &primaryAccountID
+	}
+
+	amountX := s.buildPaymentAmountHelper(ctx, bpc, bpaArg)
+	res.AmountErrMsg = amountX.amountErrMsg
+	res.WorthDescription = amountX.worthDescription
+	res.WorthInfo = amountX.worthInfo
+	readyChecklist.amount = amountX.haveAmount
```

Comment:
this unnecessarily dies if it can't show the price in XLM, but that's probably worth having less code 👍

Reply:
Before this line it was failing the check all the time, so I just added it without know much of what's behind the scene. What should the desired behavior be here? Should `readyToRequest` maybe not depend on `readyChecklist.amount`?

Reply:
There's nothing behind the scenes of readyChecklist. That's just some booleans. readyToRequest is supposed to mean that passing the args to `makeRequestLocal` would likely succeed. So `readyChecklist.amount = amountX.haveAmount` is good. I'd just leave it. I was just mentioning that if you put 3 USD and it failed to figure out how many XLM that was that would cause readyToSend to be false which is not strictly necessary.